File: ainopay-server/internal/models/payment.go
package models

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type Payment struct {
	ID              uuid.UUID      `gorm:"type:uuid;primary_key" json:"id"`
	UserID          uuid.UUID      `gorm:"type:uuid;not null" json:"user_id"`
	User            User           `gorm:"foreignKey:UserID" json:"user,omitempty"`
	Amount          float64        `gorm:"type:decimal(15,2);not null" json:"amount"`
	Status          string         `gorm:"type:varchar(20);default:'pending'" json:"status"` // pending, completed, failed, refunded
	PaymentMethodID uuid.UUID      `gorm:"type:uuid;not null" json:"payment_method_id"`
	PaymentMethod   PaymentMethod  `gorm:"foreignKey:PaymentMethodID" json:"payment_method,omitempty"`
	CategoryID      uuid.UUID      `gorm:"type:uuid;not null" json:"category_id"`
	Category        Category       `gorm:"foreignKey:CategoryID" json:"category,omitempty"`
	Description     string         `gorm:"type:text" json:"description"`
	TransactionDate time.Time      `gorm:"not null" json:"transaction_date"`
	CreatedAt       time.Time      `json:"created_at"`
	UpdatedAt       time.Time      `json:"updated_at"`
}

func (p *Payment) BeforeCreate(tx *gorm.DB) error {
	if p.ID == uuid.Nil {
		p.ID = uuid.New()
	}
	return nil
}

type MonthlyStats struct {
	Month       string  `json:"month"`
	TotalAmount float64 `json:"total_amount"`
	Count       int64   `json:"count"`
}

File: app/types/payment.ts
export interface Payment {
  id: string;
  user_id: string;
  amount: number;
  status: "pending" | "completed" | "failed" | "refunded";
  payment_method_id: string;
  payment_method?: PaymentMethod;
  category_id: string;
  category?: Category;
  description: string;
  transaction_date: string;
  created_at: string;
  updated_at: string;
}

export interface PaymentMethod {
  id: string;
  name: string;
  code: string;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface Category {
  id: string;
  name: string;
  description: string;
  created_at: string;
  updated_at: string;
}

export interface CreatePaymentRequest {
  amount: number;
  payment_method_id: string;
  category_id: string;
  description: string;
  transaction_date: string;
}

export interface UpdatePaymentRequest {
  amount: number;
  status: "pending" | "completed" | "failed" | "refunded";
  payment_method_id: string;
  category_id: string;
  description: string;
  transaction_date: string;
}

export interface PaymentListResponse {
  payments: Payment[];
  total: number;
  page: number;
  limit: number;
}

export interface DashboardStats {
  total_payments: number;
  completed_count: number;
  pending_count: number;
  total_amount: number;
}
